docs/faq.md

```diff
@@ -49,3 +49,36 @@ intermediates and outputs parallel with the `src` folder and not inside it,
 you can help optimize Git command performance for developers in the repository
 by limiting the number of files Git needs to consider for many common
 operations.
+
+### Why is Scalar a C# tool instead of built directly into Git?
+
+Scalar does a few things that Git currently has no precedent of doing and a few
+things that Git will never do.
+
+Git does not have a concept of running background jobs, which is critical to
+keeping our repositories running smoothly. Git does not have a concept of
+cache-servers, so we need to use the GVFS protocol during `scalar clone` to
+set up the cache-server URL. Scalar also creates the `.scalarCache` directory
+so multiple enlistments share objects from the remote, allowing a second
+`scalar clone` operation to be much faster than the first. All of these concepts
+are on our backlog for contributing to Git, in which case we can transition
+from the Scalar implementation to the Git implementation.
+
+Scalar uses the GVFS protocol for some repositories. This protocol was created
+by Azure Repos to support large repositories before the Git protocol supported
+partial clone and the Git client supported missing objects. In a way, the GVFS
+protocol proved that there was value in creating the partial clone feature.
+
+Scalar does currently require working with a
+[custom version of Git](https://github.com/microsoft/git), but we hope to relax
+that restriction eventually.
+We try to keep our fork of Git as close to [upstream](https://github.com/git-for-windows/git)
+as we can, but some features are necessary _for now_. Still, maintaining those
+features as we take updates from upstream has a significant cost. It is better
+to create features in C# that use Git as a black-box, then remove those C#
+features as we contribute replacements to Git upstream.
+
+Finally, as Git gains the features that we need in Scalar, Scalar will need to
+support backwards compatibility with repositories cloned with older versions of
+Scalar. Using the C# layer, we can update the Git version and create a transition
+plan for old repositories to whatever new features are added to Git.

```

Comment:
`create a transition plan` reads very passively. I think you mean something more like `provide backwards compatibility` or `upgrade`, where Scalar will make sure old repos can take advantage of the new features without recloning?